card/lib/card/query.rb

```diff
@@ -97,8 +97,7 @@ def initialize statement
     # Card::Fetch)
 
     def self.run statement
-      query = new statement
-      query.run
+      new(statement.deep_clone).run
```

Comment:
I predict that if we deep clone before the initialization we won't have the same test failures.  We'll see ;)